pyiceberg/table/__init__.py

```diff
@@ -472,7 +473,11 @@ class SetLocationUpdate(TableUpdate):
 
 class SetPropertiesUpdate(TableUpdate):
     action: TableUpdateAction = TableUpdateAction.set_properties
-    updates: Dict[str, str]
+    updates: Properties
```

Comment:
I think this one should be `Dict[str, str]` because at the time when the model is constructed, the before validator has converted all values to str